daemon/meson.build

```diff
@@ -55,13 +53,47 @@ daemon_cpp_db_path_arg = '-DDB_PATH="' + serviceDBPath + '"'
 serviceDBKeyPrefix = get_option('service-db-key-prefix')
 daemon_cpp_db_key_prefix_arg = '-DMESON_KEY_PREFIX="' + serviceDBKeyPrefix + '"'
 
-nns_ml_agent_executable = executable('machine-learning-agent',
-  nns_ml_agent_srcs,
-  dependencies: [ai_service_daemon_deps],
-  include_directories: nns_ml_agent_incs,
+ml_agentd_lib_srcs = [ml_agentd_lib_common_srcs, ml_agentd_lib_service_db_srcs]
+ml_agentd_shared_lib = shared_library ('ml-agentd',
+  ml_agentd_lib_srcs,
+  dependencies: ml_agentd_deps,
+  include_directories: ml_agentd_incs,
   install: true,
-  install_dir: api_install_bindir,
+  install_dir: api_install_libdir,
   cpp_args: [daemon_cpp_db_path_arg, daemon_cpp_db_key_prefix_arg],
+  version: api_version,
+)
+ml_agentd_static_lib = static_library('ml-agentd',
+  ml_agentd_lib_srcs,
+  dependencies: ml_agentd_deps,
+  include_directories: ml_agentd_incs,
+  install: true,
+  install_dir: api_install_libdir,
+  cpp_args: [daemon_cpp_db_path_arg, daemon_cpp_db_key_prefix_arg],
+  pic: true,
+)
+
+ml_agentd_lib = ml_agentd_shared_lib
+if get_option('default_library') == 'static'
+    ml_agentd_lib = ml_agentd_static_lib
+endif
+
```

Comment:
Suggestion: define dependency to ml-agentd library here, and use it in ml_agent_executable and unittest.
```
ml_agentd_lib_dep = declare_dependency(
  link_with: ml_agentd_lib,
  dependencies: ml_agentd_deps,
  include_directories: ml_agentd_incs
)
```